test/helpers/helpers.go

```diff
@@ -456,6 +475,50 @@ func RestartMarketplace(client test.FrameworkClient, namespace string) error {
 	return nil
 }
 
+// ScaleDownMarketplace scales the marketplace deployment down to zero
+func ScaleDownMarketplace(client test.FrameworkClient, namespace string) error {
+	marketplace := &apps.Deployment{}
+	err := client.Get(context.TODO(), types.NamespacedName{Name: "marketplace-operator", Namespace: namespace}, marketplace)
+	if err != nil {
+		return err
+	}
+	zero := int32(0)
+	marketplace.Spec.Replicas = &zero
+	err = client.Update(context.TODO(), marketplace)
+	if err != nil {
+		return err
+	}
+	// Wait for deployment to scale down
+	err = WaitForDeploymentScaled(client, "marketplace-operator", namespace, zero)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// ScaleUpMarketplace scales the marketplace deployment up to one
+func ScaleUpMarketplace(client test.FrameworkClient, namespace string) error {
```

Comment:
What do you think about merging this and the above functions into a single `ScaleMarketplace` function that takes in the scale factor?

Reply:
A good abstraction is to have a lower level helper that takes scale as an input but have higher level ScaleUp and ScaleDown wrappers that call this function.

Reply:
I like that idea!

Reply:
Yeah, thinking about this more there should just be one function `ScaleMarketplace` that just takes the number of replicas. Then to scale up and down you call `ScaleMarketplace(0)` wait for it to return then call `ScaleMarketplace(1)`. Then the `RestartMarketplace` method can just call scale the second time with the previous replica size in order to preserve the behavior.